### ISSUE TITLE
ClassCastException in GitCheckoutStepHandler after update to 2.8.0-rc1

### ISSUE THREAD

**Issue by sithmein** [jenkinsci/opentelemetry-plugin#467]
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  
  ```text
Jenkins: 2.332.3
OS: Linux - 4.14.276-211.499.amzn2.x86_64
---
ace-editor:1.1
amazon-ecr:1.73.v741d474abe74
ansicolor:1.0.1
ant:475.vf34069fef73c
antisamy-markup-formatter:2.7
apache-httpcomponents-client-4-api:4.5.13-1.0
artifactory:3.16.2
atlassian-jira-software-cloud:2.0.2
authentication-tokens:1.4
aws-credentials:191.vcb_f183ce58b_9
aws-java-sdk:1.12.215-339.vdc07efc5320c
aws-java-sdk-cloudformation:1.12.215-339.vdc07efc5320c
aws-java-sdk-codebuild:1.12.215-339.vdc07efc5320c
aws-java-sdk-ec2:1.12.215-339.vdc07efc5320c
aws-java-sdk-ecr:1.12.215-339.vdc07efc5320c
aws-java-sdk-ecs:1.12.215-339.vdc07efc5320c
aws-java-sdk-elasticbeanstalk:1.12.215-339.vdc07efc5320c
aws-java-sdk-iam:1.12.215-339.vdc07efc5320c
aws-java-sdk-logs:1.12.215-339.vdc07efc5320c
aws-java-sdk-minimal:1.12.215-339.vdc07efc5320c
aws-java-sdk-ssm:1.12.215-339.vdc07efc5320c
azure-credentials:216.ve0b_4a_485ffc2
azure-sdk:106.v552de1e64d56
basic-branch-build-strategies:1.3.2
bitbucket:223.vd12f2bca5430
bitbucket-push-and-pull-request:2.8.1
blueocean:1.25.5
blueocean-autofavorite:1.2.5
blueocean-bitbucket-pipeline:1.25.5
blueocean-commons:1.25.5
blueocean-config:1.25.5
blueocean-core-js:1.25.5
blueocean-dashboard:1.25.5
blueocean-display-url:2.4.1
blueocean-events:1.25.5
blueocean-git-pipeline:1.25.5
blueocean-github-pipeline:1.25.5
blueocean-i18n:1.25.5
blueocean-jwt:1.25.5
blueocean-personalization:1.25.5
blueocean-pipeline-api-impl:1.25.5
blueocean-pipeline-editor:1.25.5
blueocean-pipeline-scm-api:1.25.5
blueocean-rest:1.25.5
blueocean-rest-impl:1.25.5
blueocean-web:1.25.5
bootstrap4-api:4.6.0-5
bootstrap5-api:5.1.3-7
bouncycastle-api:2.26
branch-api:2.1046.v0ca_37783ecc5
build-monitor-plugin:1.13+build.202205140447
build-timeout:1.21
caffeine-api:2.9.3-65.v6a_47d0f4d1fe
checks-api:1.7.4
cli-commander:0.3
cloudbees-bitbucket-branch-source:773.v4b_9b_005b_562b_
cloudbees-disk-usage-simple:0.10
cloudbees-folder:6.722.v8165b_a_cf25e9
command-launcher:84.v4a_97f2027398
conditional-buildstep:1.4.2
config-file-provider:3.10.0
configuration-as-code:1429.v09b_044a_c93de
copyartifact:1.46.4
credentials:1087.1089.v2f1b_9a_b_040e4
credentials-binding:523.vd859a_4b_122e6
dashboard-view:2.432.va_712ce35862d
dependency-check-jenkins-plugin:5.1.2
depgraph-view:1.0.5
disk-usage:0.28
display-url-api:2.3.6
docker-commons:1.19
docker-java-api:3.2.13-37.vf3411c9828b9
docker-plugin:1.2.9
docker-workflow:1.28
downstream-build-cache:1.7
durable-task:496.va67c6f9eefa7
ec2-cloud-plugin:1.2.16.0054-19708a1
echarts-api:5.3.2-2
ecs-cloud-plugin:0.9.7.0020-c4e4b86
email-ext:2.88
extended-read-permission:3.2
external-monitor-job:191.v363d0d1efdf8
favorite:2.4.1
font-awesome-api:6.1.1-1
git:4.11.3
git-client:3.11.0
git-server:1.11
github:1.34.3
github-api:1.303-400.v35c2d8258028
github-branch-source:1637.vd833b_7ca_7654
gradle:1.38
greenballs:1.15.1
h2-api:1.4.199
handy-uri-templates-2-api:2.1.8-22.v77d5b_75e6953
htmlpublisher:1.30
ivy:2.2
jackson2-api:2.13.3-285.vc03c0256d517
javadoc:217.v905b_86277a_2a_
javax-activation-api:1.2.0-3
javax-mail-api:1.6.2-6
jaxb:2.3.6-1
jdk-tool:1.5
jenkins-design-language:1.25.5
jira:3.7.1
jira-steps:2.0.141.vd0c6e6dc83f0
jjwt-api:0.11.2-71.v2722b_b_06a_2a_f
jmh-report:0.9.0
jnr-posix-api:3.1.7-3
job-dsl:1.79
job-exporter:0.4
jobConfigHistory:1139.v888b_656ca_f6d
jquery:1.12.4-1
jquery3-api:3.6.0-4
jsch:0.1.55.2
junit:1119.va_a_5e9068da_d7
junit-attachments:101.v82f494a_00e9e
ldap:2.10
mailer:414.vcc4c33714601
material-theme:0.4.1
matrix-auth:3.1.2
matrix-project:771.v574584b_39e60
maven-plugin:3.18
mercurial:2.16.2
metrics:4.1.6.2
monitoring:1.91.0
node-iterator-api:1.5.1
okhttp-api:4.9.3-105.vb96869f8ac3a
opentelemetry:2.7.1-rc2
pam-auth:1.8
parameterized-trigger:2.44
permissive-script-security:0.7
pipeline-build-step:2.18
pipeline-graph-analysis:195.v5812d95a_a_2f9
pipeline-groovy-lib:591.v3a_7f422b_d058
pipeline-input-step:448.v37cea_9a_10a_70
pipeline-maven:3.10.0
pipeline-milestone-step:101.vd572fef9d926
pipeline-model-api:2.2086.v12b_420f036e5
pipeline-model-definition:2.2086.v12b_420f036e5
pipeline-model-extensions:2.2086.v12b_420f036e5
pipeline-multibranch-defaults:2.1
pipeline-rest-api:2.24
pipeline-stage-step:293.v200037eefcd5
pipeline-stage-tags-metadata:2.2086.v12b_420f036e5
pipeline-utility-steps:2.12.1
plain-credentials:1.8
plugin-util-api:2.17.0
popper-api:1.16.1-3
popper2-api:2.11.5-2
postbuildscript:3.1.0-375.v3db_cd92485e1
prometheus:2.0.11
pubsub-light:1.16
resource-disposer:0.19
rich-text-publisher-plugin:1.4
run-condition:1.5
run-selector:1.1.1
scm-api:608.vfa_f971c5a_a_e9
script-security:1175.v4b_d517d6db_f0
shelve-project-plugin:3.2
snakeyaml-api:1.30.1
sonar:2.14
sse-gateway:1.25
ssh-agent:295.v9ca_a_1c7cc3a_a_
ssh-credentials:277.v95c2fec1c047
ssh-slaves:1.814.vc82988f54b_10
sshd:3.237.v883d165a_c1d3
structs:318.va_f3ccb_729b_71
theme-manager:0.6
timestamper:1.17
token-macro:293.v283932a_0a_b_49
trilead-api:1.57.v6e90e07157e1
uno-choice:2.6.1
variant:1.4
view-job-filters:2.3
windows-slaves:1.8.1
workflow-api:1153.vb_912c0e47fb_a_
workflow-basic-steps:948.v2c72a_091b_b_68
workflow-cps:2706.v71dd22b_c5a_a_2
workflow-durable-task-step:1139.v252a_e12e8463
workflow-job:1182.v60a_e6279b_579
workflow-multibranch:716.vc692a_e52371b_
workflow-scm-step:400.v6b_89a_1317c9a_
workflow-step-api:625.vd896b_f445a_f8
workflow-support:820.vd1a_6cc65ef33
ws-cleanup:0.42
yet-another-build-visualizer:1.16
  ```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

We are using the Docker image on Amazon Linux 2.

### Reproduction steps

1. Install/update OpenTelemetry plug-in to 2.8.0-rc1
2. Restart Jenkins
3. Wait for some jobs to start

### Expected Results

No exceptions related to the OpenTelemetry in the controller log.

### Actual Results

The following ClassCastException frequently occurs in the controller's log:

```
2022-06-30 08:56:12.772+0000 [id=129]   WARNING i.j.p.o.j.j.GraphListenerAdapterToPipelineListener#fireOnBeforeAtomicStep: Exception invoking `onBeforeAtomicStep` on TracingPipelineListener{}
java.lang.ClassCastException: class org.jenkinsci.plugins.structs.describable.UninstantiatedDescribable cannot be cast to class java.util.Map (org.jenkinsci.plugins.structs.describable.UninstantiatedDescribable is in unnamed module of loader jenkins.util.AntClassLoader @277f6b6f; java.util.Map is in module java.base of loader 'bootstrap')
        at io.jenkins.plugins.opentelemetry.job.step.GitCheckoutStepHandler.canCreateSpanBuilder(GitCheckoutStepHandler.java:57)
        at io.jenkins.plugins.opentelemetry.job.MonitoringPipelineListener.lambda$onAtomicStep$5(MonitoringPipelineListener.java:194)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
        at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
        at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline.findFirst(Unknown Source)
        at io.jenkins.plugins.opentelemetry.job.MonitoringPipelineListener.onAtomicStep(MonitoringPipelineListener.java:194)
        at io.jenkins.plugins.opentelemetry.job.jenkins.GraphListenerAdapterToPipelineListener.fireOnBeforeAtomicStep(GraphListenerAdapterToPipelineListener.java:212)
        at io.jenkins.plugins.opentelemetry.job.jenkins.GraphListenerAdapterToPipelineListener.onNewHead(GraphListenerAdapterToPipelineListener.java:75)
        at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.notifyListeners(CpsFlowExecution.java:1498)
        at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.notifyNewHead(CpsThreadGroup.java:492)
        at org.jenkinsci.plugins.workflow.cps.FlowHead.setNewHead(FlowHead.java:157)
        at org.jenkinsci.plugins.workflow.cps.DSL.invokeStep(DSL.java:288)
        at org.jenkinsci.plugins.workflow.cps.DSL.invokeMethod(DSL.java:196)
        at org.jenkinsci.plugins.workflow.cps.CpsScript.invokeMethod(CpsScript.java:124)
        at jdk.internal.reflect.GeneratedMethodAccessor627.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:98)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1225)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1034)
        at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:41)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:163)
        at org.kohsuke.groovy.sandbox.GroovyInterceptor.onMethodCall(GroovyInterceptor.java:23)
        at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:158)
        at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:161)
        at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:165)
        at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:135)
        at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:135)
        at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.methodCall(SandboxInvoker.java:17)
        at com.cloudbees.groovy.cps.impl.ContinuationGroup.methodCall(ContinuationGroup.java:86)
        at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.dispatchOrArg(FunctionCallBlock.java:113)
        at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.fixArg(FunctionCallBlock.java:83)
        at jdk.internal.reflect.GeneratedMethodAccessor502.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
        at com.cloudbees.groovy.cps.impl.PropertyishBlock$ContinuationImpl.get(PropertyishBlock.java:76)
        at com.cloudbees.groovy.cps.LValueBlock$GetAdapter.receive(LValueBlock.java:30)
        at com.cloudbees.groovy.cps.impl.PropertyishBlock$ContinuationImpl.fixName(PropertyishBlock.java:66)
        at jdk.internal.reflect.GeneratedMethodAccessor499.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
        at com.cloudbees.groovy.cps.impl.ConstantBlock.eval(ConstantBlock.java:21)
        at com.cloudbees.groovy.cps.Next.step(Next.java:83)
        at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:174)
        at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:163)
        at org.codehaus.groovy.runtime.GroovyCategorySupport$ThreadCategoryInfo.use(GroovyCategorySupport.java:136)
        at org.codehaus.groovy.runtime.GroovyCategorySupport.use(GroovyCategorySupport.java:275)
        at com.cloudbees.groovy.cps.Continuable.run0(Continuable.java:163)
        at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.access$001(SandboxContinuable.java:18)
        at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.run0(SandboxContinuable.java:51)
        at org.jenkinsci.plugins.workflow.cps.CpsThread.runNextChunk(CpsThread.java:187)
        at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:420)
        at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$400(CpsThreadGroup.java:95)
        at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:330)
        at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:294)
        at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:67)
        at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
        at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:139)
        at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
        at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:68)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
```

**Comment by cyrille-leclerc** [comment]
Fixed by
* #468 

Please try fix https://github.com/jenkinsci/opentelemetry-plugin/releases/tag/opentelemetry-2.8.0-rc2

@sithmein Could you please try the fix enabling FINE log level on `io.jenkins.plugins.opentelemetry.job.step.GitCheckoutStepHandler` and capturing the log message `Skip unexpected 'scm' type: ...` ?

I'm not sure this fragility was introduced recently in the Jenkins Otel plugin.

**Comment by sithmein** [comment]
After updated to 2.8.0-rc3 the problem seems to be gone.
I set the log level to FINE but didn't see the message yet. It only occurred very infrequently, though.

**Comment by cyrille-leclerc** [comment]
Thanks for the great news, i hope your problems are solved.

**Comment by cyrille-leclerc** [comment]
Fixed in 2.9.0
